src/quemb/shared/helper.py

```diff
@@ -66,3 +71,34 @@ def ncore_(z: int) -> int:
         print("exiting", flush=True)
         sys.exit()
     return nc
+
+
+def write_cube(
+    be_object: object, cube_file_path: PathLike, fragment_idx: Optional[list], **kwargs
+) -> None:
+    """Write cube files of embedding orbitals from a BE object.
+
+    Parameters
+    ----------
+    be_object : object
+        BE object containing the fragments, each of which contains embedding orbitals.
+    cube_file_path : PathLike
+        Directory to write the cube files to.
+    fragment_idx : Optional[list]
+        Index of the fragments to write the cube files for. If None, all fragments are written.
+    kwargs: Any
+        Keyword arguments passed to cubegen.orbital.
+    """
+    assert type(be_object) is molbe.BE, NotImplementedError(
+        "Support for Periodic BE not implemented yet."
+    )
+    if not fragment_idx:
+        fragment_idx = range(be_object.Nfrag)
+    for idx in fragment_idx:
+        for emb_orb_idx in range(be_object.Fobjs[idx].TA.shape[1]):
+            orbital(
+                be_object.mol,
+                os.path.join(cube_file_path, f"frag_{idx}_orb_{emb_orb_idx}.cube"),
```

Comment:
If cube_file_path is a `pathlib.Path` one can simply write
```python3
cube_file_path / f"frag_{idx}_orb_{emb_orb_idx}.cube"
```
But I understand if you want to keep a more tolerant interface.

Reply:
I may have missed something, but `PathLike` could either be `pathlib.Path` or `str`. The tolerant interface is necessary for `str`, right?

Reply:
Yes, if you want to also accept strings you have to either use `os.path.join` or narrow the type down with an explicit cast to `Path`.

Note how the constructor [here](https://github.com/troyvvgroup/quemb/blob/main/src/quemb/shared/manage_scratch.py#L47)
accepts `PathLike`, but then quickly casts to `Path` and internally it is only `Path` as seen [here](https://github.com/troyvvgroup/quemb/blob/main/src/quemb/shared/manage_scratch.py#L44).
This then allows nicer working with pure `Path` as shown [here](https://github.com/troyvvgroup/quemb/blob/main/src/quemb/shared/manage_scratch.py#L108)

My comment is really just a comment; if you want to keep it internally as `PathLike`, go for it.